pysmiles/write_smiles.py

```diff
@@ -187,3 +189,38 @@ def keyfunc(idx):
 
     smiles += ')' * branch_depth
     return smiles
+
+def write_smiles(molecule, default_element='*', start=None):
+    """
+    Creates a SMILES string describing `molecule` according to the OpenSMILES
+    standard. If `molecule` consists of multiple disconnected components their
+    corresponding SMILES string will be joined by zero-order bonds (".").
+
+    Parameters
+    ----------
+    molecule : nx.Graph
+        The molecule for which a SMILES string should be generated.
+    default_element : str
+        The element to write if the attribute is missing for a node.
+    start : Hashable
+        The atom at which the depth first traversal of the molecule should
+        start. A sensible one is chosen: preferably a terminal heteroatom.
+
+    Returns
+    -------
+    str
+        The SMILES string describing `molecule`.
+    """
+    smiles = []
+    for nodes in nx.connected_components(molecule):
```

Comment:
```suggestion
    all_connected_components = sorted(nx.connected_components(molecule), key=str)     # make order deterministic
    for nodes in all_connected_components:
```

Reply:
Just noticed my previous one wouldn't have worked well. This one should be better. It sorts by the str of sorted elements in the graph component. The same graph should now nearly always have the same order of fragments and therefore the same smiles. If this is too involved, it might be better to leave it out though.
```suggestion
    element_str = lambda nodes: str(sorted([molecule.nodes[idx].get('element', default_element) for idx in nodes])) 
    all_connected_components = sorted(nx.connected_components(molecule), key=element_str)     # make order deterministic```